allensdk/brain_observatory/behavior/behavior_ophys_session.py

```diff
@@ -162,15 +163,39 @@ def get_performance_metrics(self):
         performance_metrics['mean_dprime'] = rolling_performance_df['dprime'].mean()
         performance_metrics['mean_dprime_engaged'] = rolling_performance_df['dprime'][engaged_trial_mask].mean()
 
+    def get_records(self, tables=None):
 
+        if tables is None:
+            tables = ['trials', 'dff_traces', 'corrected_fluorescence_traces']
+
+        m = [{'running_speed': self.running_speed.values, 'timestamps': self.running_speed.timestamps}]
+        for table in tables:
+            df = getattr(self, table)
+            m += df.to_dict(orient='records')
+        return m
+
+    def query(self, query, tables=None, **kwargs):
+
+        records = self.get_records(tables)
+        data = [x for x in records if query.is_satisfied_by(x)]
+
+        return pd.DataFrame.from_records(data, **kwargs)
         
 
-        for key, val in performance_metrics.items():
-            print(key, val)
 
 
 if __name__ == "__main__":
 
-    ophys_experiment_id = 789359614
+    ophys_experiment_id = 792813858
     session = BehaviorOphysSession.from_lims(ophys_experiment_id)
     session.get_performance_metrics()
+
+    # df = session.query(Or(IsRunningSpeed(), CellRoiId(793298517)).And(TimestampInterval(10, 10.06)))
+    # print(df)
+
+    df = session.query(Equals(True, 'hit'), columns=['start_time'])
+    start_times = df['start_time'].values[:5]
+    stop_times = start_times + .1
+
+    df = session.query(And(IsRunningSpeed().Not(), CellRoiId(793298517)).And(StackedIntervals(start_times, stop_times)))
```

Comment:
Would be really cool if you could use operator overloading to replace basic operators like `AND`, `NOT`, and `OR` (maybe with `&`, `~`, and `|`) (see: https://www.python-course.eu/python3_magic_methods.php)